MBBSEmu/Session/LocalConsole/LocalConsoleSession.cs

```diff
@@ -0,0 +1,95 @@
+                0x0090, 0x0091, 0x0092, 0x0093, 0x0094, 0x0095, 0x0096, 0x0097, 0x0098, 0x0099, 0x009A, 0x009B, 0x009C, 0x009D, 0x009E, 0x009F,
+                0x00A0, 0x00A1, 0x00A2, 0x00A3, 0x00A4, 0x00A5, 0x00A6, 0x00A7, 0x00A8, 0x00A9, 0x00AA, 0x00AB, 0x00AC, 0x00AD, 0x00AE, 0x00AF,
+                0x2591, 0x2592, 0x2593, 0x2503, 0x251C, 0x2561, 0x2562, 0x2556, 0x2555, 0x2563, 0x2551, 0x2557, 0x255D, 0x255C, 0x255B, 0x2510,
+                0x2514, 0x2534, 0x252C, 0x251C, 0x2500, 0x253C, 0x255E, 0x255F, 0x255A, 0x2554, 0x2569, 0x2566, 0x2560, 0x2550, 0x256C, 0x6725,
+                0x2568, 0x2564, 0x2565, 0x2559, 0x2558, 0x2552, 0x2553, 0x256B, 0x256A, 0x2518, 0x250C, 0x2588, 0x2584, 0x258C, 0x2590, 0x2580,
+                0x03B1, 0x04F1, 0x03C0, 0x06A3, 0x00E4, 0x00E5, 0x00E6, 0x00E7, 0x00E8, 0x00E9, 0x00EA, 0x00EB, 0x00EC, 0x00ED, 0x00EE, 0x00EF,
+                0x00F0, 0x00F1, 0x00F2, 0x00F3, 0x00F4, 0x00F5, 0x00F6, 0x00F7, 0x00F8, 0x00F9, 0x00FA, 0x00FB, 0x00FC, 0x00FD, 0x00FE, 0x00FF
+        };
+
+        public LocalConsoleSession(string sessionId, IMbbsHost host) : base(sessionId)
+        {
+            _logger = ServiceResolver.GetService<ILogger>();
+            SendToClientMethod = dataToSend => UnicodeANSIOutput(dataToSend);
+            Console.Clear();
+            Console.OutputEncoding = Encoding.Unicode;
+            SessionState = EnumSessionState.Unauthenticated;
+
+            (_logger as CustomLogger)?.DisableConsoleLogging();
+
+            _consoleInputThreadIsRunning = true;
+            _consoleInputThread = new Thread(InputThread);
+            _consoleInputThread.Start();
+            host.AddSession(this);
+        }
+
+        private void InputThread()
+        {
+            while (SessionState != EnumSessionState.LoggedOff && _consoleInputThreadIsRunning)
+            {
+                DataFromClient.Add((byte)Console.ReadKey(true).KeyChar);
+                ProcessDataFromClient();
+            }
+        }
+
+        /// <summary>
+        ///     Takes an input Extended ASCII string and converts it to Unicode Output
+        /// </summary>
+        /// <param name="inputString"></param>
+        /// <returns></returns>
+        private void UnicodeANSIOutput(ReadOnlySpan<byte> inputString)
+        {
+            foreach (var c in inputString)
+            {
+
+                //Standard ASCII Characters are written as-is
+                if (c <= 127)
+                {
+                    Console.Write((char)c);
+                    continue;
+                }
+
+                //Extended ASCII Characters are converted to their Unicode Counterparts
+                Console.Write(Encoding.Unicode.GetString(BitConverter.GetBytes(_extendedASCIItoUnicode[c])));
+            }
+        }
+
+        public override void Stop()
+        {
+            _consoleInputThreadIsRunning = false;
+            Environment.Exit(0);
```

Comment:
this will prevent modules from shutting down cleanly no?

Reply:
Let me see if I can invoke the Ctrl-C behavior...

OR we could always send them back to the login, so it's always running on the console until they Ctrl-C. This way someone could run MBBSEmu in a screen and just jump to the screen when they want to play vs. telnetting.

Reply:
Modified the method to Call `Stop()` on the `MbbsHost` before exiting, this will give modules a chance to run their shutdown routines.